crates/ty_python_semantic/resources/mdtest/annotations/new_types.md

```diff
@@ -167,6 +167,48 @@ on top of that:
 Foo = NewType("Foo", 42)
 ```
 
+## `float` and `complex` special cases
+
+`float` and `complex` are subject to a special case in the typing spec, which we currently interpret
+to mean that `float` in type position is `int | float`, and `complex` in type position is
+`int | float | complex`. This is awkward for `NewType`, because as we just tested above, unions
+aren't valid `NewType` bases. However, `float` and `complex` _are_ valid `NewType` bases. We need to
+avoid interpreting them as unions for the purpose of inferring a base, and yet continue interpreting
+them as unions for the purpose of typechecking an instantiation.
+
+```py
+from typing import NewType
+
+Foo = NewType("Foo", float)
+Foo(42)
+Foo(3.14)
+Foo("hello")  # error: [invalid-argument-type] "Argument is incorrect: Expected `int | float`, found `Literal["hello"]`"
```

Comment:
Would it be valuable to expand the happy-path tests here with e.g. assertions about `__class__`, subtyping, etc.? Speaking of which, it's not actually clear to me whether `Foo` here should be considered a subtype of `float`...